cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -473,5 +477,48 @@ public Program create(String path) throws IOException {
       return Programs.createWithUnpack(location, Files.createTempDir());
     }
   }
+
+  /**
+   * A {@link SecurityManager} used by the runnable container. It specifically disabling
+   * Spark classes to call {@link System#exit(int)}, {@link Runtime#halt(int)}
+   * and {@link System#setSecurityManager(SecurityManager)}.
+   * This is to workaround modification done in HDP Spark (CDAP-3014).
+   */
+  private static final class RunnableSecurityManager extends SecurityManager {
+    @Override
+    public void checkPermission(Permission perm) {
+      if ("setSecurityManager".equals(perm.getName()) && isFromSpark()) {
```

Comment:
oh ok. so it needs to be suppressed completely. so do we become future proof if other systems also decide to call ``System.exit()``? :)